src/components/ToDoApp.js

```diff
@@ -1,17 +1,20 @@
-import React, { Component } from 'react';
 import '../App.css';
-import { ToDoList } from './ToDoList';
 import { Input } from './Input';
+import { ToDoList } from './ToDoList';
+import React, { Component } from 'react';
 import { Navigation } from './Navigation';
+import { ALL_TODOS, REMAINING_TODOS, COMPLETED_TODOS } from '../constants/constants';
 
 export class ToDoApp extends Component {
 
   constructor(props) {
     super(props);
+    const storageTodos = window.localStorage.getItem('storageTodos');
```

Comment:
Not in a `constructor` as well :D Maybe `componentDidMount`?

Reply:
Won't that require a call to setState(), which causes rerendering at the beginning?

According to [docs](https://reactjs.org/docs/react-component.html#componentdidmount)

> You may call setState() immediately in componentDidMount(). It will trigger an extra rendering, but it will happen before the browser updates the screen. **This guarantees that even though the render() will be called twice in this case, the user won’t see the intermediate state. Use this pattern with caution because it often causes performance issues. In most cases, you should be able to assign the initial state in the constructor() instead.** It can, however, be necessary for cases like modals and tooltips when you need to measure a DOM node before rendering something that depends on its size or position.